tests/e2e/default/enketo/training-cards.wdio-spec.js

```diff
@@ -0,0 +1,117 @@
+const fs = require('fs');
+
+const utils = require('../../../utils');
+const loginPage = require('../../../page-objects/default/login/login.wdio.page');
+const commonPage = require('../../../page-objects/default/common/common.wdio.page');
+const trainingCardsPage = require('../../../page-objects/default/enketo/training-cards.wdio.page');
+const placeFactory = require('../../../factories/cht/contacts/place');
+const userFactory = require('../../../factories/cht/users/users');
+const personFactory = require('../../../factories/cht/contacts/person');
+const commonElements = require('../../../page-objects/default/common/common.wdio.page');
+const reportsPage = require('../../../page-objects/default/reports/reports.wdio.page');
+
+describe('Training Cards', () => {
+  const parent = placeFactory.place().build({ _id: 'dist1', type: 'district_hospital' });
+  const user = userFactory.build({ roles: [ 'nurse', 'chw' ] });
+  const patient = personFactory.build({ parent: { _id: user.place._id, parent: { _id: parent._id } } });
+  const formDoc = {
+    _id: 'form:training:text_only',
+    internalId: 'training:text_only',
+    title: 'Text Only Training',
+    type: 'form',
+    context: {
+      start_date: new Date().getTime(),
+      user_roles: [ 'nurse' ],
+      duration: 5,
+    },
+    _attachments: {
+      xml: {
+        content_type: 'application/octet-stream',
+        data: Buffer
+          .from(fs.readFileSync(`${__dirname}/forms/training-cards-text-only.xml`, 'utf8'))
+          .toString('base64'),
+      },
+    },
+  };
+
+  let savedFormDoc;
+  before(async () => {
+    await utils.saveDocs([ parent, patient ]);
+    savedFormDoc = await utils.saveDoc(formDoc);
+    await utils.createUsers([ user ]);
+    await loginPage.login(user);
+    await commonElements.waitForPageLoaded();
+  });
+
+  it('should cancel training', async () => {
+    await commonPage.goToMessages();
+    await trainingCardsPage.waitForTrainingCards();
+
+    const confirmMessage = await trainingCardsPage.quitTraining();
+    expect(confirmMessage.header).to.equal('Important changes');
+    expect(confirmMessage.body).to.contain(
+      'This training is not finished. You will lose your progress if you leave now. Are you sure you want to leave?'
+    );
+    await trainingCardsPage.confirmQuitTraining();
+
+    await commonPage.goToReports();
+    await commonElements.waitForPageLoaded();
+    expect(await reportsPage.allReports()).to.be.empty;
+  });
+
+  it('should display training after form was updated', async () => {
+    await commonPage.goToMessages();
+    await commonElements.waitForPageLoaded();
+
```

Comment:
Is the training card visible here already or not?

Reply:
What I meant was that after the user cancels the training, it will reappear automatically when the form is changed.

Reply:
Yes, the previous test case cancels the training card when this test runs. 
But I agree it's not easy to read, so I've improved this test: 
- reload to get the training to appear
- cancel the training again 
- push an update to the training
- check that training displays again.

Reply:
[Commit](https://github.com/medic/cht-core/pull/8017/commits/46aefb92d609a5bca9240beae474d24aa8cf809d)